apiserver/src/main/java/org/dependencytrack/persistence/jdbi/FindingDao.java

```diff
@@ -182,6 +187,8 @@ record GroupedFindingRow(
                AND "COMPONENT"."PROJECT_ID" = "A"."PROJECT_ID"
               INNER JOIN "PROJECT"
                 ON "COMPONENT"."PROJECT_ID" = "PROJECT"."ID"
+              LEFT JOIN "COMPONENT_OCCURRENCE" co
+                ON "COMPONENT"."ID" = "co"."COMPONENT_ID"
```

Comment:
This `LEFT JOIN` can significantly increase the result set size. 

Consider using an `EXISTS` subquery instead in the `SELECT` clause, e.g.:

```sql
EXISTS(SELECT 1 FROM "COMPONENT_OCCURRENCE" WHERE "COMPONENT_ID" = "COMPONENT"."ID") AS "componentHasOccurrences"
```